### PR TITLE
Modify report code to receive last changed param for bamboo api

### PR DESCRIPTION
**Pull request by AgostinaDev**
## Description

- Custom reports changes that were made in commit efeed5c9c027b5ff54cee8f55d9005b3fa8d202b were returning data in a string format (different from the format it was being returned before the changes, so it was not working properly.
- This change is needed for a Forecast feature.
- This is the BambooHR API documentation reference: 
  - https://documentation.bamboohr.com/reference/request-custom-report-1

## Related Ticket
- [[BE] Create task to keep people rates updated](https://rootstrap.atlassian.net/browse/FBRSB-2530)

## Screenshots (if appropriate):
Before these changes:
imagen

After these changes:
imagen

Local tests run result:
imagen

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by claudiokerekes — APPROVED** [review]
LGTM

File: lib/bamboozled/api/report.rb [hunk 1]
```diff
@@ -6,30 +6,36 @@ def find(number, format = "JSON", fd_param = true)
...
       private
 
-      def body(fields, include_null, date)
+      def body(fields, include_null, last_changed_date)
         body = {}
-        body[:fields] = fields || FieldCollection.all_names
-        last_changed_object = {}
-        last_changed_object[:includeNull] = include_null.to_s
-        last_changed_object[:value] = date
+        body[:fields] = if fields == :all
```
Comment (claramaffei) [inline lib/bamboozled/api/report.rb:23]: Maybe the if clause could be in a separate method. Just a suggestion. No blocker 😊

**Review by claramaffei — APPROVED** [review]
LGTM!

File: lib/bamboozled/api/report.rb (hunk 1 above)
Comment (AgostinaDev) [reply lib/bamboozled/api/report.rb:23]: Thanks for the suggestion! I'll pass this time and prioritize functionality given the context of this project

**Pull request merged**